indexing-service/src/test/java/org/apache/druid/indexing/overlord/ForkingTaskRunnerTest.java

```diff
@@ -537,6 +539,78 @@ ProcessHolder runTaskProcess(List<String> command, File logFile, TaskLocation ta
     Assert.assertTrue(forkingTaskRunner.restore().isEmpty());
   }
 
+  @Test
+  public void testTaskRunDoesntPassLoadLookupsArgument() throws Exception
+  {
+    verifyTaskRunLoadLookupsArgument(LookupLoadingMode.ALL);
+  }
+
+  @Test
+  public void testTaskRunPassLoadLookupsArgument() throws Exception
+  {
+    verifyTaskRunLoadLookupsArgument(LookupLoadingMode.NONE);
+  }
+
+  private void verifyTaskRunLoadLookupsArgument(LookupLoadingMode lookupLoadingMode) throws Exception
+  {
+    ObjectMapper mapper = new DefaultObjectMapper();
+    Task task = new NoopTask(null, null, null, 0, 0, null) {
+      @Override
+      public LookupLoadingMode loadLookups()
+      {
+        return lookupLoadingMode;
+      }
+    };
+    File file = temporaryFolder.newFolder();
+    TaskConfig taskConfig = makeDefaultTaskConfigBuilder()
+        .setBaseTaskDir(file.toString())
+        .build();
+    final WorkerConfig workerConfig = new WorkerConfig();
+    ForkingTaskRunner forkingTaskRunner = new ForkingTaskRunner(
+        new ForkingTaskRunnerConfig(),
+        taskConfig,
+        workerConfig,
+        new Properties(),
+        new NoopTaskLogs(),
+        mapper,
+        new DruidNode("middleManager", "host", false, 8091, null, true, false),
+        new StartupLoggingConfig(),
+        TaskStorageDirTracker.fromConfigs(workerConfig, taskConfig)
+    )
+    {
+      @Override
+      ProcessHolder runTaskProcess(List<String> command, File logFile, TaskLocation taskLocation) throws IOException
+      {
+        if (lookupLoadingMode == LookupLoadingMode.NONE) {
+          Assert.assertTrue(command.containsAll(ImmutableSet.of("--loadLookups", LookupLoadingMode.NONE.name())));
+        } else {
+          Assert.assertFalse(command.containsAll(ImmutableSet.of("--loadLookups", LookupLoadingMode.NONE.name())));
```

Comment:
Nit:

This assertion should be broken up to ensure that neither of the values are present.

```suggestion
          Assert.assertFalse(command.contains("--loadLookups"));
          Assert.assertFalse(command.contains(LookupLoadingMode.NONE.name()));
```